internal/runners/clean/run_lin_mac.go

```diff
@@ -146,3 +136,58 @@ func removeInstall(cfg configurable) error {
 func verifyInstallation() error {
 	return nil
 }
+
+var errDirNotEmpty = errs.New("Not empty")
+
+func removeEmptyDir(dir string) error {
+	empty, err := fileutils.IsEmptyDir(dir)
+	if err == nil && empty {
+		removeErr := os.RemoveAll(dir)
+		if err != nil {
+			return errs.Wrap(removeErr, "Could not remove directory")
+		}
+	} else if err != nil {
+		return errs.Wrap(err, "Could not check if directory is empty")
+	}
+
+	if !empty {
+		return errDirNotEmpty
+	}
+
+	return nil
+}
+
+func cleanInstallDir(dir string) error {
+	var asFiles = []string{
+		installation.InstallDirMarker,
+		constants.StateInstallerCmd + exeutils.Extension,
+
+		// Remove all of the state tool executables and finally the
+		// bin directory
+		filepath.Join(installation.BinDirName, appinfo.StateApp().Exec()),
+		filepath.Join(installation.BinDirName, appinfo.SvcApp().Exec()),
+		filepath.Join(installation.BinDirName, appinfo.TrayApp().Exec()),
+		installation.BinDirName,
+
+		// The system directory is on MacOS only and contains the tray
+		// application files. It is safe for us to remove this directory
+		// without first inspecting the contents.
+		"system",
+	}
+
+	for _, file := range asFiles {
+		f := filepath.Join(dir, file)
+
+		var err error
+		if fileutils.DirExists(f) && fileutils.IsDir(f) {
+			err = os.RemoveAll(f)
```

Comment:
Hmm we should probably do the empty check here as well.

Reply:
We're relying on the order of the slice above rather than using the empty check. The `system` directory is not empty. For that, we would have to walk the directory and delete the application files or have a representation of the directory structure somewhere to refer to.